site/content/protocols/action-menu/2.0/readme.md

```diff
@@ -0,0 +1,206 @@
+
+The protocol can be initiated by either the `requester` asking for the root menu or the `responder` sending an unsolicited root menu. The protocol ends when the `requester` issues a perform operation or an internal timeout on the `responder` causes it to discard menu context. At any time a `requester` can reset the protocol by requesting the root menu from a `responder`.
+
+Whilst the protocol is defined here as uni-directional (i.e requester to responder), both Agents may support both requester and responder roles simultaneously. Such cases would result in two instances of the action-menu protocol operating in parrallel.
+
+## States
+
+[![state machines](state-machines.png)](https://docs.google.com/spreadsheets/d/1KZ78KDxmXdck068aJpn-7BbDNfeaTcUyDgk6b2efCxc/edit?usp=sharing)
+
+#### States for Requester
+
+State&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;| Description
+------ | -----------
+null | No _menu_ has been requested or received
+awaiting-root-menu | _menu-request_ message has been sent and awaiting root _menu_ response
+preparing-selection | _menu_ message has been received and a user selection is pending
+done | _perform_ message has been sent and protocol has finished. Perform actions can include requesting a new menu which will re-enter the state machine with the receive-menu event from the null state.
+
+#### States for Responder
+
+State&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;| Description
+------ | -----------
+null | No _menu_ has been requested or sent
+preparing-root-menu | _menu-request_ message has been received and preparing _menu_ response for root menu
+awaiting-selection | _menu_ message has been sent and are awaiting a _perform_ request
+done | _perform_ message has been received and protocol has finished. Perform actions can include requesting a new menu which will re-enter the state machine with the send-menu event from the null state.
+
+
+## Design By Contract
+
+No protocol specific errors exist. Any errors related to headers or other core features are documented in the appropriate places.
+
+## Security
+
+This protocol expects messages to be encrypted during transmission, and repudiable. 
+
+## Composition
+
+Supported Goal Code | Notes
+--- | ---
+                     |       
+                     |       
+
+
+
+## Message Reference
+
+### Menu
+A `requestor` is expected to display only one active menu per connection when action menus are employed by the `responder`. A newly received menu is not expected to interrupt a user, but rather be made available for the user to inspect possible actions related to the `responder`.
```

Comment:
Roles are defined as `requester` and  `responder`. In 1.0 protocol, requester role wording was a bit mixed up and seems to be fixed here, but this `requestor` is still there.

Reply:
fixed